lib/awful/widget/taglist.lua

```diff
@@ -365,55 +367,93 @@ local function taglist_update(s, w, buttons, filter, data, style, update_functio
+-- @tparam[opt=nil] string args.style.squares_sel_empty
+-- @tparam[opt=nil] string args.style.squares_unsel_empty
+-- @tparam[opt=nil] string args.style.squares_resize
+-- @tparam[opt=nil] string args.style.disable_icon
+-- @tparam[opt=nil] string args.style.font
+-- @tparam[opt=nil] number args.style.spacing The spacing between tags.
+-- @tparam[opt] string args.style.squares_sel A user provided image for selected squares.
+-- @tparam[opt] string args.style.squares_unsel A user provided image for unselected squares.
+-- @tparam[opt] string args.style.squares_sel_empty A user provided image for selected squares for empty tags.
+-- @tparam[opt] string args.style.squares_unsel_empty A user provided image for unselected squares for empty tags.
+-- @tparam[opt] boolean args.style.squares_resize True or false to resize squares.
+-- @tparam string|pattern args.style.bg_focus The background color for focused client.
+-- @tparam string|pattern args.style.fg_focus The foreground color for focused client.
+-- @tparam string|pattern args.style.bg_urgent The background color for urgent clients.
+-- @tparam string|pattern args.style.fg_urgent The foreground color for urgent clients.
+-- @tparam string args.style.font The font.
+-- @param filter **DEPRECATED** use args.filter
+-- @param buttons **DEPRECATED** use args.buttons
+-- @param style **DEPRECATED** use args.style
+-- @param update_function **DEPRECATED** use args.update_function
+-- @param base_widget **DEPRECATED** use args.base_widget
 -- @function awful.taglist
-function taglist.new(screen, filter, buttons, style, update_function, base_widget)
-    screen = get_screen(screen)
-    local uf = update_function or common.list_update
-    local w = base_widget or fixed.horizontal()
+function taglist.new(args, filter, buttons, style, update_function, base_widget)
 
-    if w.set_spacing and (style and style.spacing or beautiful.taglist_spacing) then
-        w:set_spacing(style and style.spacing or beautiful.taglist_spacing)
+    local screen = nil
+
+    local argstype = type(args)
+
+    -- Detect the old function signature
+    if argstype == "number" or argstype == "screen" or
+      (argstype == "table" and args.index and args == capi.screen[args.index]) then
+        gdebug.deprecate("The `screen` paramater is deprecated, use `args.screen`.",
+            {deprecated_in=5})
+
+        screen = get_screen(args)
+        args = {}
+    end
+
+    assert(type(args) == "table")
+
+    local deprecated_param = {filter, buttons, style, update_function, base_widget}
+    local deprecated_names = {"filter", "buttons", "style",
+                              "update_function", "layout"}
+
+    for k=1, 5 do
+        local v = deprecated_param[k]
+        if v then
+            gdebug.deprecate("The `awful.widget.taglist()` `"..deprecated_names[k]
+                .."` paramater is deprecated, use `args."..deprecated_names[k].."`.",
+            {deprecated_in=5})
+            args[deprecated_names[k]] = v
```

Comment:
Umm. I guess so... However given this can only affect the users who *both* have migrated to the new `args` syntax *and* still have deprecated parameters, I would argue nobody will ever be realistically affected. What do you think?